api/envoy/api/v2/route/route.proto

```diff
@@ -553,24 +553,38 @@ message RouteAction {
   // backend).
   repeated HashPolicy hash_policy = 15;
 
-  // Indicates that a HTTP/1.1 client connection to this particular route
-  // should be allowed (and expected) to upgrade to a WebSocket connection. The
-  // default is false.
+  // Indicates that a HTTP/1.1 client connection to this particular route is allowed to
+  // upgrade to a WebSocket connection. The default is false.
   //
   // .. attention::
   //
-  //   If set to true, Envoy will expect the first request matching this route to
-  //   contain WebSocket upgrade headers. If the headers are not present, the
-  //   connection will be rejected. If set to true, Envoy will setup plain TCP
+  //   If a connection is upgraded to a WebSocket connection, Envoy will setup plain TCP
   //   proxying between the client and the upstream server. Hence, an upstream
   //   server that rejects the WebSocket upgrade request is also responsible for
   //   closing the associated connection. Until then, Envoy will continue to
   //   proxy data from the client to the upstream server.
   //
-  //   Redirects, timeouts and retries are not supported on routes where websocket upgrades are
-  //   allowed.
+  //   Redirects are not supported on routes where WebSocket upgrades are allowed.
   google.protobuf.BoolValue use_websocket = 16;
 
+  message WebSocketProxyConfig {
+    // See :ref:`idle_timeout <envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.stat_prefix>`.
+    // If the parameter is not specified, the default value of "websocket" is used.
+    string stat_prefix = 1;
+
+    // See :ref:`idle_timeout <envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.idle_timeout>`.
+    google.protobuf.Duration idle_timeout = 2
+      [(validate.rules).duration.gt = {}, (gogoproto.stdduration) = true];
+
+    // See :ref:`max_connect_attempts
+    // <envoy_api_field_config.filter.network.tcp_proxy.v2.TcpProxy.max_connect_attempts>`.
+    google.protobuf.UInt32Value max_connect_attempts = 3 [(validate.rules).uint32.gte = 1];
```

Comment:
Do you think we could add a http request timeout here as well ? (Future). This is for the first http upgrade request/resp

Reply:
I'd rather not add it until someone is going to implement it.

What operation do you want to have governed by this timeout? Right now, as soon as we connect to the upstream (in the websocket/tcpproxy case) the tcpproxy idle_timeout is in effect. Are you wanting a different timeout for the intial http upgrade response?

Reply:
yes, because there is no timeout for the initial upgrade response.. or may be its the default 15s